homeassistant/components/neptune_apex/light.py

```diff
@@ -0,0 +1,131 @@
+"""Support outlet status/control from Neptune Apex."""
+import logging
+
+import pynepsys
+
+from homeassistant.components.light import ATTR_EFFECT, SUPPORT_EFFECT, Light
+
+from . import NEPTUNE_APEX, NEPTUNE_APEX_COORDINATOR
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Parse through the Apex outlets and creates Outlet entities."""
+    apex = hass.data[NEPTUNE_APEX]
+    coordinator = hass.data[NEPTUNE_APEX_COORDINATOR]
+
+    async_add_entities(
+        Outlet(coordinator, apex, name)
+        for i, name in enumerate(coordinator.data.outlets)
+    )
+
+
+class Outlet(Light):
+    """Shoehorn the Apex outlet concept into a hass Light entity.
+
+    This feels gross, but the more you play with Apex, you realize that once you get past
+    a certain point of revulsion, there is beauty in it. And madness.
+    """
+
+    def __init__(self, coordinator, apex: pynepsys.Apex, name):
+        """Initialize this outlet, storing reference to parent Apex."""
+        self.coordinator = coordinator
+        self._name = name
+        self.apex = apex
+        self.outlet: pynepsys.Outlet = apex.outlets[name]
+
+    @property
+    def name(self):
+        """Name of this outlet (From the Apex), prefixed."""
+        return "apex." + self._name
+
+    @property
+    def unique_id(self):
+        """Name of this outlet as defined in the Apex."""
+        return self.name
```

Comment:
Can the outlet name be changed by the user?
```suggestion
        return self._name
```

Reply:
I wrote this a while ago, I think I was just trying to avoid repeated code where both 'name' and 'unique_id' would be 'apex.'+ outlet name. So from previous PR change, I have the Apex serial #... Should I be adding that to this unique id? Am I doing this all wrong?

Reply:
I would use the serial number + the name of the outlet as unique id, as long as the user can't change the name of the outlet.